### DISCUSSION TITLE
Is this still supported?  Pyupdater is no longer being maintained

### DISCUSSION THREAD

**Discussion by ghost** [dennisvang/tufup#114] [Q&A]
Is this still supported? [Pyupdater is no longer being maintained][1].

[1]: https://github.com/Digital-Sapphire/PyUpdater#this-is-the-end

**Comment by dennisvang** [comment]
@Hardcorekila Yes, `tufup` is still supported.

In fact, `tufup` was created **because** PyUpdater is no longer maintained.

`tufup` provides the same basic functionality as PyUpdater, but with better (potential) security through the use of [TUF][1].

The readme provides [instructions for migrating from PyUpdater to tufup][2].

[1]: https://theupdateframework.io/
[2]: https://github.com/dennisvang/tufup/blob/master/README.md#migrating-from-other-update-frameworks

**Comment by dennisvang** [comment]
Do note that, although `tufup` offers the same *basic* functionality as PyUpdater, there are some important differences:

- `tufup` uses `python-tuf` to secure the update process. This means you'll need to understand a little bit about the principles behind [TUF][1]. You can tailor the level of security to your needs, from highly secure, to slightly secure.
- PyUpdater was tightly integrated with [PyInstaller][2], but `tufup` is not. You *can* use `tufup` with PyInstaller, as illustrated in the [tufup-example][3], but it is not required. You can also use it with any other packaging method, or you can even distribute a plain python script, or just any directory of files.
- PyUpdater's handling of release channels was [broken][4], `tufup` should handle them just fine.

[1]: https://theupdateframework.io/overview/
[2]: https://pyinstaller.org/en/stable/
[3]: https://github.com/dennisvang/tufup-example
[4]: https://github.com/Digital-Sapphire/PyUpdater/issues/255

**Comment by ghost** [comment]
@dennisvang  I seem to be getting errors from the example would my best bet be to just use pyupdater which I am familiar with and migrate from that?

```
C:\Users\vinni\PycharmProjects\pythonProject3\venv\Scripts\python.exe C:\Users\vinni\PycharmProjects\pythonProject3\pyupdater.py 
Traceback (most recent call last):
  File "C:\Users\vinni\PycharmProjects\pythonProject3\pyupdater.py", line 8, in <module>
    from tufup.repo import (
  File "C:\Users\vinni\AppData\Local\Programs\Python\Python310\lib\site-packages\tufup\__init__.py", line 4, in <module>
    from tufup.repo import cli
  File "C:\Users\vinni\AppData\Local\Programs\Python\Python310\lib\site-packages\tufup\repo\__init__.py", line 9, in <module>
    import setuptools.config.expand  # noqa
ModuleNotFoundError: No module named 'setuptools.config.expand'; 'setuptools.config' is not a package

Process finished with exit code 1
```

**Comment by dennisvang** [comment]
@Hardcorekila No you should be able to run the example. This error is caused by an old version of setuptools.

It looks like you haven't installed all the dev requirements from the `tufup-example` repo:

`pip install -r requirements-dev.txt`

Also see [tufup-example readme][1].

[1]: https://github.com/dennisvang/tufup-example#setup

**Comment by ghost** [comment]
@dennisvang  Any chance of a video tutorial with an example that incorporates say Bitbucket or Github in the future?  I.e Updating 1.0 -> 1.1 and then 1.1 to 1.2

**Comment by dennisvang** [comment]
@Hardcorekila Unfortunately, setting up workflows for Bitbucket or Github, like configuring servers, is outside the scope of `tufup`. It's just too specific to your implementation, and there are too many different ways to do it.

The tufup-example repo illustrates how the basic components of the update system interact. How to implement this in production is up to the user.

**Comment by dennisvang** [comment]
@Hardcorekila Just curious: Did you get the example up and running?
